dashboard/modules/metrics/metrics_head.py

```diff
@@ -0,0 +1,62 @@
+import logging
+import os
+import shutil
+
+from ray.dashboard.modules.metrics.grafana_datasource_template import (
+    GRAFANA_DATASOURCE_TEMPLATE,
+)
+import ray.dashboard.utils as dashboard_utils
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+PROMETHEUS_HOST_ENV_VAR = "RAY_PROMETHEUS_HOST"
+DEFAULT_PROMETHEUS_HOST = "http://localhost:9090"
+METRICS_PATH = "/tmp/ray/metrics"
+GRAFANA_CONFIG_OUTPUT_PATH = f"{METRICS_PATH}/grafana"
+GRAFANA_CONFIG_INPUT_PATH = os.path.join(os.path.dirname(__file__), "grafana")
+PROMETHEUS_CONFIG_OUTPUT_PATH = f"{METRICS_PATH}/prometheus/prometheus.yml"
+PROMETHEUS_CONFIG_INPUT_PATH = os.path.join(
+    os.path.dirname(__file__), "prometheus", "prometheus.yml"
+)
+
+
+class MetricsHead(dashboard_utils.DashboardHeadModule):
+    def __init__(self, dashboard_head):
+        super().__init__(dashboard_head)
+
+    @staticmethod
+    def is_minimal_module():
+        return False
+
+    async def run(self, server):
+        # Copy default grafana configurations
+        if os.path.exists(GRAFANA_CONFIG_OUTPUT_PATH):
+            shutil.rmtree(GRAFANA_CONFIG_OUTPUT_PATH)
+        os.makedirs(os.path.dirname(GRAFANA_CONFIG_OUTPUT_PATH), exist_ok=True)
+        shutil.copytree(GRAFANA_CONFIG_INPUT_PATH, GRAFANA_CONFIG_OUTPUT_PATH)
```

Comment:
What happens if users provide their own config?

Reply:
Maybe we need an API for users to provide their own config (e.g., add a template to a specific folder).

Reply:
I'll create a folder `/tmp/ray/metrics/grafana/custom-dashboards`. Any dashboard json found there can be automatically picked up. We won't ever delete that folder.

Reply:
Is it going to be in this PR?

Reply:
yes.. in progress